github/Repository.py

```diff
@@ -2732,6 +2734,153 @@ def get_repository_advisory(self, ghsa: str) -> github.RepositoryAdvisory.Reposi
         headers, data = self._requester.requestJsonAndCheck("GET", f"{self.url}/security-advisories/{ghsa}")
         return github.RepositoryAdvisory.RepositoryAdvisory(self._requester, headers, data)
 
+    def get_rules_for_branch(self, branch: str) -> PaginatedList[github.Rule.Rule]:
+        """
+        :calls: `GET /repos/{owner}/{repo}/rules/branches/{branch} <https://docs.github.com/en/rest/repos/rules#get-rules-for-a-branch>`_
+        :param branch: str
+        :rtype: :class:`PaginatedList` of :class:`github.Rule.Rule`
+        """
+        assert isinstance(branch, str), branch
+        branch = urllib.parse.quote(branch)
+        return PaginatedList(
+            github.Rule.Rule,
+            self._requester,
+            f"{self.url}/rules/branches/{branch}",
+            None,
+        )
+
+    def get_rulesets(
+        self, includes_parents: bool = True, targets: Opt[list[str]] = NotSet
+    ) -> PaginatedList[github.Ruleset.Ruleset]:
+        """
+        :calls: `GET /repos/{owner}/{repo}/rulesets <https://docs.github.com/en/rest/repos/rules#get-all-repository-rulesets>`_
+        :param includes_parents: bool
+        :param targets: list of string
+        :rtype: :class:`PaginatedList` of :class:`github.Ruleset.Ruleset`
+        """
+        assert isinstance(includes_parents, bool), includes_parents
+        assert is_optional_list(targets, str), targets
+        url_parameters: dict[str, Any] = {"includes_parents": includes_parents}
+        if is_defined(targets):
+            url_parameters["targets"] = ",".join(targets)
+        return PaginatedList(
+            github.Ruleset.Ruleset,
+            self._requester,
+            f"{self.url}/rulesets",
+            url_parameters,
+        )
+
+    def get_ruleset(self, ruleset_id: int) -> github.Ruleset.Ruleset:
+        """
+        :calls: `GET /repos/{owner}/{repo}/rulesets/{ruleset_id} <https://docs.github.com/en/rest/repos/rules#get-a-repository-ruleset>`_
+        :param ruleset_id: int
+        :rtype: :class:`github.Ruleset.Ruleset`
+        """
+        assert isinstance(ruleset_id, int), ruleset_id
+        attrs = {"id": ruleset_id, "url": f"{self.url}/rulesets/{ruleset_id}"}
+        return github.Ruleset.Ruleset(self._requester, attributes=attrs)
```

Comment:
This returns a lazy `Ruleset`, meaning no API call occurs. Only when accessing an attribute other than `id` or `url`, the call occurs. This allows to move `Repository.update_ruleset` to `Ruleset.update`, as that method uses only `self.url`